packages/features/bookings/Booker/Booker.tsx

```diff
@@ -0,0 +1,213 @@
+  });
+
+  useInitializeBookerStore({
+    username,
+    eventSlug,
+    month,
+    eventId: event?.data?.id,
+    rescheduleUid,
+    rescheduleBooking,
+  });
+
+  useEffect(() => {
+    setLayout(isMobile ? "mobile" : "small_calendar");
+  }, [isMobile, setLayout]);
+
+  useEffect(() => {
+    if (event.isLoading) return setBookerState("loading");
+    if (!selectedDate) return setBookerState("selecting_date");
+    if (!selectedTimeslot) return setBookerState("selecting_time");
+    return setBookerState("booking");
+  }, [event, selectedDate, selectedTimeslot, setBookerState]);
+
+  useEffect(() => {
+    if (layout === "mobile") {
+      timeslotsRef.current?.scrollIntoView({ behavior: "smooth" });
+    }
+  }, [layout, selectedDate]);
+
+  return (
+    <>
+      {/*
+        If we would render this on mobile, it would unset the mobile variant,
+        since that's not a valid option, so it would set the layout to null.
+      */}
+      {!isMobile && (
+        <div className="[&>div]:bg-muted fixed top-2 right-3 z-10">
+          <ToggleGroup
+            onValueChange={(layout) => setLayout(layout as BookerLayout)}
+            defaultValue="small_calendar"
+            options={[
+              {
+                value: "small_calendar",
+                label: <Calendar width="16" height="16" />,
+                tooltip: t("switch_monthly"),
+              },
+              {
+                value: "large_calendar",
+                label: <Grid width="16" height="16" />,
+                tooltip: t("switch_weekly"),
+              },
+              {
+                value: "large_timeslots",
+                label: <Columns width="16" height="16" />,
+                tooltip: t("switch_multiday"),
+              },
+            ]}
+          />
+        </div>
+      )}
+      <div className="flex h-full w-full flex-col items-center">
+        <m.div
+          layout
+          // Passing the default animation styles here as the styles, makes sure that there's no initial loading state
+          // where there's no styles applied yet (meaning there wouldn't be a grid + widths), which would cause
+          // the layout to jump around on load.
+          style={resizeAnimationConfig.small_calendar.default as MotionStyle}
+          animate={resizeAnimationConfig[layout]?.[bookerState] || resizeAnimationConfig[layout].default}
+          transition={{ ease: "easeInOut", duration: 0.4 }}
+          className={classNames(
+            "[--booker-meta-width:280px] [--booker-main-width:480px] [--booker-timeslots-width:240px] lg:[--booker-timeslots-width:280px]",
+            "bg-muted grid max-w-full items-start overflow-clip dark:[color-scheme:dark] md:flex-row",
+            layout === "small_calendar" &&
+              "border-subtle mt-20 min-h-[450px] w-[calc(var(--booker-meta-width)+var(--booker-main-width))] rounded-md border",
+            layout !== "small_calendar" && "h-auto min-h-screen w-screen"
+          )}>
```

Comment:
Im not against all these arbitrary values but as much as they are being used here would it not be easier to do this in style tag?

Reply:
Youre not wrong its deffo neater :P